packages/catalog-platformstate-writer/src/consumerServiceV2.ts

```diff
@@ -1,12 +1,202 @@
-  _dynamoDBClient: DynamoDBClient
+  dynamoDBClient: DynamoDBClient
 ): Promise<void> {
   await match(message)
+    .with({ type: "EServiceDescriptorPublished" }, async (msg) => {
+      const { eservice, descriptor } = parseEServiceAndDescriptor(
+        msg.data.eservice,
+        unsafeBrandId(msg.data.descriptorId)
+      );
+      const previousDescriptor = eservice.descriptors.find(
+        (d) => d.version === (Number(descriptor.version) - 1).toString()
+      );
+
+      // flow for current descriptor
+      {
+        const primaryKeyCurrent = makePlatformStatesEServiceDescriptorPK({
+          eserviceId: eservice.id,
+          descriptorId: descriptor.id,
+        });
+        const existingCatalogEntryCurrent = await readCatalogEntry(
+          primaryKeyCurrent,
+          dynamoDBClient
+        );
+        if (
+          existingCatalogEntryCurrent &&
+          existingCatalogEntryCurrent.version > msg.version
+        ) {
+          // Stops processing if the message is older than the catalog entry
+          return;
+        } else if (
+          existingCatalogEntryCurrent &&
+          existingCatalogEntryCurrent.version <= msg.version
+        ) {
+          await updateDescriptorStateInPlatformStatesEntry(
+            dynamoDBClient,
+            primaryKeyCurrent,
+            descriptorStateToClientState(descriptor.state),
+            msg.version
+          );
+
+          // token-generation-states
+          const eserviceId_descriptorId = makeGSIPKEServiceIdDescriptorId({
+            eserviceId: eservice.id,
+            descriptorId: descriptor.id,
+          });
+          await updateDescriptorStateInTokenGenerationStatesTable(
+            eserviceId_descriptorId,
+            descriptor.state,
+            dynamoDBClient
+          );
+        } else {
+          const catalogEntry: PlatformStatesCatalogEntry = {
+            PK: primaryKeyCurrent,
+            state: descriptorStateToClientState(descriptor.state),
+            descriptorAudience: descriptor.audience[0],
+            version: msg.version,
+            updatedAt: new Date().toISOString(),
+          };
+
+          await writeCatalogEntry(catalogEntry, dynamoDBClient);
+
+          // token-generation-states
+          const eserviceId_descriptorId = makeGSIPKEServiceIdDescriptorId({
+            eserviceId: eservice.id,
+            descriptorId: descriptor.id,
+          });
+          await updateDescriptorStateInTokenGenerationStatesTable(
+            eserviceId_descriptorId,
+            descriptor.state,
+            dynamoDBClient
+          );
+        }
+      }
+
+      // flow for previous descriptor
+
+      if (
+        !previousDescriptor ||
+        previousDescriptor.state !== descriptorState.archived
+      ) {
+        return Promise.resolve();
+      } else {
+        const primaryKeyPrevious = makePlatformStatesEServiceDescriptorPK({
+          eserviceId: eservice.id,
+          descriptorId: descriptor.id,
```

Comment:
may be worth looking on tests for this
```suggestion
          descriptorId: previousDescriptor.id,
```

Reply:
Fixed, thanks!